### ISSUE TITLE
Folder environment variables don't merge into sub and base environments; override instead

### ISSUE THREAD

**Issue by anno1985** [getinsomnia/insomnia#2822]
**Describe the bug**
When you define a "variable tree" in your base environment, you can selectively override it by setting some of it in your sub environments. Sub environments are effectively merged into base environments. Variables defined in folder environments will completely override the variable tree, however. This is unexpected (and unhelpful).

**To Reproduce**
Steps to reproduce the behaviour:
1.  Create a **New Workspace** (called "Environment Variables")
2. Set its **Base Environment** to the following:
```json
{
  "foo": "foo",
  "bar": "bar",
  "baz": "baz",
  "middle": {
    "foo": "foo",
    "bar": "bar",
    "baz": "baz",
    "bottom": {
      "foo": "foo",
      "bar": "bar",
      "baz": "baz"
    }
  },
  "query": "{{ foo }},{{ middle.foo }},{{ middle.bottom.foo }},{{ bar }},{{ middle.bar }},{{ middle.bottom.bar }},{{ baz }},{{ middle.baz }},{{ middle.bottom.baz }}"
}
```
3. Create a **new folder** My Project
4. Create a new **GET request** to http://localhost?query={{query}}

5. Create a new **Sub Environment** "My Env":
```json
{
  "bar": "env",
  "middle": {
    "bar": "env",
    "bottom": {
      "bar": "env"
    }
  }
}
```
4. When **No Environment** is **selected**, `query` (in the request) resolves to `foo,foo,foo,bar,bar,bar,baz,baz,baz`, as expected. ✅ 
5. When **My Env** is **selected**, `query` (in the request) doesn't seem to replace its values and resolves to `{{ foo }},{{ middle.foo }},{{ middle.bottom.foo }},{{ bar }},{{ middle.bar }},{{ middle.bottom.bar }},{{ baz }},{{ middle.baz }},{{ middle.bottom.baz }}` in the live preview. **This is unexpected.** ❌ 
6. When opening the **Base Env**, the query variable there seems resolve to `{root}: foo,{root}: foo{root}: foo,{My Env}: env,{My Env}: env,{My Env}: env,{root}: baz,{root}: baz,{root}: baz` when hovering over them, as expected. (This is what I'd expect for the previous step as well.) ✅ 
7. Add the following to the **environment** of the **My Project folder**:
```json
{
  "baz": "folder",
  "middle": {
    "baz": "folder",
    "bottom": {
      "baz": "folder"
    }
  }
}
```
8. **Select My Env**
9. The live preview for the query parameter in the GET request doesn't seem to resolve: `{{ foo }},{{ middle.foo }},{{ middle.bottom.foo }},{{ bar }},{{ middle.bar }},{{ middle.bottom.bar }},{{ baz }},{{ middle.baz }},{{ middle.bottom.baz }}`. **This is unexpected.** ❌ 
10. Opening the **Base Environment**, `query` seems to resolve to
```
{root}: foo,
attempted to output null or undefined value,
attempted to output null or undefined value,
{MyEnv}: env,
attempted to output null or undefined value,
attempted to output null or undefined value,
{My Project}: folder,
{My Project}: folder,
{My Project}: folder
```
**This is unexpected.** ❌ 

It looks like the MyProject env is overriding (instead of merging) the variables it has defined (with the exception of the "root" object, which is merged - otherwise, {{foo}} and {{bar}} would resolve to an error as well). 
I expect similar issues when defining the folder level environment and deselecting the Sub Environment.

**Expected behavior**
When selecting the sub environment and setting the variables in all three environments, I would expect `query` (in the GET request, and as an example in the Base environment, too) , which is defined as
```
{{ foo }},
{{ middle.foo }},
{{ middle.bottom.foo }},
{{ bar }},
{{ middle.bar }},
{{ middle.bottom.bar }},
{{ baz }},
{{ middle.baz }},
{{ middle.bottom.baz }}
```

 to resolve to the following
```
{root}: foo,
{root}: foo,
{root}: foo,
{MyEnv}: env,
{MyEnv}: env,
{MyEnv}: env,
{My Project}: folder,
{My Project}: folder,
{My Project}: folder
```

In other words, I would expect the folder environment to use the merging strategy, just as the base environment seems to do, instead of just merging what's on the JSON root, and replacing the any and all "property trees" (as seen in step 10).

**Screenshots**
n/a

**Desktop (please complete the following information):**
 - OS: macOS 10.15.7
 - Installation Method: direct download
 - App Version 2020.4.2

**Additional context**
https://support.insomnia.rest/article/18-environment-variables

**Comment by anno1985** [comment]
How can this be both, needing triage and won't fix... ;-)
